controller/internal/routes/posture_check_router.go

```diff
@@ -66,7 +69,49 @@ func (r *PostureCheckRouter) Register(ae *env.AppEnv) {
 }
 
 func (r *PostureCheckRouter) List(ae *env.AppEnv, rc *response.RequestContext) {
-	ListWithHandler(ae, rc, ae.Handlers.PostureCheck, MapPostureCheckToRestEntity)
+	//ListWithHandler(ae, rc, ae.Handlers.PostureCheck, MapPostureCheckToRestEntity)
+	List(rc, func(rc *response.RequestContext, queryOptions *QueryOptions) (*QueryResult, error) {
+		query, err := queryOptions.getFullQuery(ae.Handlers.PostureCheck.GetStore())
+		if err != nil {
+			return nil, err
+		}
+
+		roleFilters := rc.Request.URL.Query()["roleFilter"]
+		roleSemantic := rc.Request.URL.Query().Get("roleSemantic")
+
+		var apiEntities []interface{}
+		var qmd *models.QueryMetaData
+		if len(roleFilters) > 0 {
+			cursorProvider, err := ae.GetStores().PostureCheck.GetRoleAttributesCursorProvider(roleFilters, roleSemantic)
+			if err != nil {
+				return nil, err
+			}
+
+			result, err := ae.Handlers.PostureCheck.BasePreparedListIndexed(cursorProvider, query)
+
+			if err != nil {
+				return nil, err
+			}
+
+			apiEntities, err = modelToApi(ae, rc, MapPostureCheckToRestEntity, result.GetEntities())
+			if err != nil {
+				return nil, err
+			}
+			qmd = &result.QueryMetaData
+		} else {
+			result, err := ae.Handlers.PostureCheck.QueryPostureChecks(query)
+			if err != nil {
+				pfxlog.Logger().Errorf("error executing list query: %+v", err)
```

Comment:
why is this one error logged when all the others aren't - does this need to be logged here vs the others?

Reply:
debug I left in.